File: modules/native-tts-media-control/ios/NativeTTSMediaControlModule.swift
import ExpoModulesCore
import Foundation

public class NativeTTSMediaControlModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NativeTTSMediaControl")

    Events("TTSPlay", "TTSPause", "TTSStop", "TTSPrev", "TTSNext", "TTSRewind", "TTSSeekTo")

    Function("showMediaNotification") { (title: String, subtitle: String, coverUri: String, isPlaying: Bool) in
      // Not implemented on iOS
    }

    Function("updatePlaybackState") { (isPlaying: Bool) in
      // Not implemented on iOS
    }

    Function("updateProgress") { (current: Double, total: Double) in
      // Not implemented on iOS
    }

    Function("dismiss") {
      // Not implemented on iOS
    }
  }
}

File: modules/native-tts-media-control/src/NativeTTSMediaControlModule.ts
import { requireNativeModule, NativeModule } from 'expo-modules-core';

type NativeTTSMediaControlEvents = {
  TTSPlay: () => void;
  TTSPause: () => void;
  TTSStop: () => void;
  TTSPrev: () => void;
  TTSNext: () => void;
  TTSRewind: () => void;
  TTSSeekTo: (params: { position: number }) => void;
};

declare class NativeTTSMediaControl extends NativeModule<NativeTTSMediaControlEvents> {
  showMediaNotification(
    title: string,
    subtitle: string,
    coverUri: string,
    playing: boolean,
  ): void;
  updatePlaybackState(playing: boolean): void;
  updateProgress(current: number, total: number): void;
  dismiss(): void;
}

export default requireNativeModule<NativeTTSMediaControl>('NativeTTSMediaControl');